### PR TITLE
feat(Server): Extract session metrics in non processing relays (INGEST-250)

### PR DESCRIPTION
**Pull request by RaduW**
Enable metrics extraction from sessions for all realys (processing and non processing).

INGEST-250

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 - Add sampling based on transaction name. ([#1058](https://github.com/getsentry/relay/pull/1058))
 - Support running Relay without config directory. The most important configuration, including Relay mode and credentials, can now be provided through commandline arguments or environment variables alone. ([#1055](https://github.com/getsentry/relay/pull/1055)
+- Extract session metrics in non processing relays (INGEST-250). ([#1073](https://github.com/getsentry/relay/pull/1073))
```
Comment (jan-auer) [inline CHANGELOG.md]: Let's not add JIRA tickets to the public changelog.

File: tests/integration/fixtures/mini_sentry.py [hunk 2]
```diff
@@ -119,12 +119,21 @@ def basic_project_config(self, project_id, dsn_public_key=None):
...
+                **ret_val,
+                **extra,
+                "config": {**ret_val["config"], **extra_config},
+            }
+
         self.project_configs[project_id] = ret_val
         return ret_val
 
-    def add_full_project_config(self, project_id, dsn_public_key=None):
+    def full_project_config(self, project_id, dsn_public_key=None, extra=None):
```
Comment (jan-auer) [inline tests/integration/fixtures/mini_sentry.py:136]: This is fine, but don't we usually just patch this straight away in the tests?

**Review by jan-auer — COMMENTED** [review]
Relay code changes and added test scenarios look good to me. Please also update imports in `actors/envelopes.rs`, as the removed feature gating requires imports to move, too.

I have not reviewed the test changes in detail. Regarding timeouts, `assert_empty` would usually already wait a while to ensure there are no more messages produced on Kafka.

File: CHANGELOG.md (hunk 1 above)
Comment (RaduW) [reply CHANGELOG.md]: ok

File: tests/integration/fixtures/mini_sentry.py (hunk 2 above)
Comment (RaduW) [reply tests/integration/fixtures/mini_sentry.py:136]: yes... but it is a bit odd to go inside mini_sentry's bowels to change an "already configured" project.

File: CHANGELOG.md (hunk 1 above)
Comment (RaduW) [reply CHANGELOG.md]: ok

File: tests/integration/fixtures/mini_sentry.py (hunk 2 above)
Comment (RaduW) [reply tests/integration/fixtures/mini_sentry.py:136]: yes we call `add_full_project_config` and then mess around in mini_sentry's bowels. 
I think that changing the project_config from inside mini_sentry is unnecessarily ugly.

**Comment by RaduW** [comment]
> I have not reviewed the test changes in detail. Regarding timeouts, `assert_empty` would usually already wait a while to ensure there are no more messages produced on Kafka.

`assert_empty` didn't seem to do the trick, when modifying it to test all scenarios, tests were passing that should have failed (because `assert_empty` was returning too quickly). 
I have noticed now that `assert_empty` has an optional timeout... I'll look if that was the problem.

**Review by jjbayer — APPROVED** [review]
Looks good, I fixed an unused parameter in one test, and added a test to make clear that transaction metrics are not affected.

File: tests/integration/test_metrics.py [hunk 3]
```diff
@@ -138,63 +154,197 @@ def test_metrics_full(mini_sentry, relay, relay_with_processing, metrics_consume
...
+        item_headers={"metrics_extracted": metrics_extracted},
+    )
 
     # Get session envelope
-    mini_sentry.captured_events.get(timeout=2)
+    first_envelope = mini_sentry.captured_events.get(timeout=2)
+
+    try:
+        second_envelope = mini_sentry.captured_events.get(timeout=2)
+    except:
```
Comment (untitaker) [inline tests/integration/test_metrics.py]:
```suggestion
    except Exception:
```

File: tests/integration/test_metrics.py [hunk 4]
```diff
@@ -240,70 +390,85 @@ def test_session_metrics(mini_sentry, relay_with_processing, metrics_consumer):
...
+    transaction["measurements"] = {
+        "foo": {"value": 1.2},
+        "bar": {"value": 1.3},
+    }
+    transaction["breakdowns"] = {"breakdown1": {"baz": {"value": 1.4},}}
 
-        if not feature_enabled:
-            message = metrics_consumer.poll(timeout=None)
-            assert message is None, message.value()
+    #: The `metrics_extracted` header is ignored for transactions for now.
```
Comment (untitaker) [inline tests/integration/test_metrics.py:425]: Why? Is there going to be another PR? In INGEST-250 we say we want it to work for transactions

**Review by untitaker — APPROVED** [review]

File: tests/integration/test_metrics.py (hunk 4 above)
Comment (jjbayer) [reply tests/integration/test_metrics.py:425]: There is a separate story for that: https://getsentry.atlassian.net/browse/INGEST-332.
